sootup.core/src/main/java/sootup/core/jimple/basic/LocalGenerator.java

```diff
@@ -64,138 +61,152 @@ public Local generateThisLocal(@Nonnull Type type) {
 
   /** generates a new {@link Local} given the type for field. */
   public Local generateField(@Nonnull Type type) {
-    return generate(type, true);
+    // field Locals traditionally do not begin with "$"
+
+    Local localCandidate;
+    StringBuilder name = ns.getResult();
+    // is there a name collision? retry!
+    do {
+      name.setLength(0);
+      type.accept(ns);
+      localCandidate = Jimple.newLocal(name.toString(), type);
+    } while (locals.contains(localCandidate));
+
+    locals.add(localCandidate);
+    return localCandidate;
   }
 
   /** generates a new {@link Local} given the type for local. */
   public Local generateLocal(@Nonnull Type type) {
```

Comment:
- Wouldn't it be better to use a common method for generateField and generateLocal?
- I think do while in generateField looks cleaner than what's done here
- Also what about naming generateField to generatFieldLocal? 😄

Reply:
1+2) I thought we could trade some decline of simplicity/readability for removing unnecessary runtime overhead as its not that complicated here
3) nice, I agree